src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -356,52 +358,78 @@ public void onOverlayDrawn(RenderGameOverlayEvent.Post event) {
 							world.getBlockState(Minecraft.getMinecraft().objectMouseOver.getBlockPos());
 						Item matchItem = Item.getItemFromBlock(match.getBlock());
 						if (matchItem != null) {
-							ItemStack matchStack = new ItemStack(matchItem, 1,
+							boolean ruler = heldInternal.equals("BUILDERS_RULER");
+							ItemStack matchStack;
+
+							matchStack = new ItemStack(matchItem, 1,
 								match
 									.getBlock()
 									.getDamageValue(
 										world,
 										Minecraft.getMinecraft().objectMouseOver.getBlockPos()
 									)
 							);
+							if (ruler) {
+								if (!Minecraft.getMinecraft().thePlayer.isSneaking()) matchStack = getFirstItemInRuler();
+								match = Blocks.dirt.getDefaultState();
+							}
+							if (matchStack == null) return;
 
-							getBuildersWandCandidates(
-								Minecraft.getMinecraft().thePlayer,
-								Minecraft.getMinecraft().objectMouseOver,
-								event.partialTicks,
-								candidatesOld,
-								candidatesOldSorted,
-								999 - MAX_BUILDERS_BLOCKS
-							);
+							if (heldInternal.equals("BUILDERS_WAND")) {
```

Comment:
This is too many indentation levels. Anymore than 4-5 and you should be worried. This is insanity